### ISSUE TITLE
Write-Output strings are posted multiple times in the "Test Job" window.

### ISSUE THREAD

**Issue by justinmetz** [azureautomation/azure-automation-ise-addon#107]
It looks like a single Write-Output call is echoed to the Test Job output window multiple times.  An example of this behavior shown between the test job output and a local txt log of the same runbook instance is attached.
![test_job_output](https://cloud.githubusercontent.com/assets/12280312/14252090/ce1527b0-fa53-11e5-906b-a601dceddfaf.jpg)

**Comment by eamonoreilly** [comment]
Hi Justin,
Can you confirm what build you are on so I can make sure I'm investigating with the right source. Can you click on the help button and there should be a version on the top.
Thanks,
Eamon

**Comment by justinmetz** [comment]
Hi Eamon,

Here's the info from the help tab..

Release informationVisit the Github Commit Page
https://github.com/azureautomation/azure-automation-ise-addon/commits/masterfor
latest improvements.

Updates for release 0.2.3.2
- Added certificate support
- Test console streams output without refreshing screen
- Warn if local verison is newer than cloud version when testing draft

Thanks,

Justin

**Comment by eamonoreilly** [comment]
Thanks Justin. I'll have a look. Could you verify one more thing for me? If you look at this job in the Azure portal, does it have the duplicate lines also or is it correctly displaying there? This will let me know if it is an engine issue within the service or the ISE Add-On that is doing something incorrect.
Thanks again,
Eamon

**Comment by justinmetz** [comment]
Here's the output in the Azure portal test console...  Seems to work in there..
![job_output_2](https://cloud.githubusercontent.com/assets/12280312/14258665/6e03af90-fa71-11e5-9728-178ecdac4fd3.jpg)

**Comment by JoeBrockhaus** [comment]
FWIW, I've seen this a number of times as well, but have never been able to pinpoint it (slash, never had the time to spend trying to repro). 

I have assumed that it has to do with how the ISE plugin makes the numerous requests to stream the log data back to the client, and then adds them to the display. If data is coming back in batches, and once batch overlaps another (because more data is being added from the job), it would make sense that the plugin code may be ignoring the overlap?

That said, I can't rule out having seen it in the web ui. 

Generally, I want to say that when I did see it, it was typically one of the first messages to get written to output. IE: if doing some processing in a loop, I might see the first iteration's output twice, but then all the remaining output wouldn't dupe. 

Unfortunately I also am not sure if closing then re-opening the test job window or the test pane makes this duplication go away, or if even I saw the same duplication between the web ui and plugin (ie: once the job has completed; if it did, that would seem to indicate it's an engine thing, perhaps.)

I know that's all not very helpful and circumstantial (at best), but just to say I've seen it intermittently under these circumstances :)

HTH :100::dash:
